server/services/main.js

```diff
@@ -65,6 +66,33 @@ module.exports = () => ({
     fs.unlinkSync(`${strapi.config.get('plugin.config-sync.syncDir')}${configName}.json`);
   },
 
+  /**
+   * Zip config files.
+   *
+   * @param {string} configName - The name of the config file.
+   * @returns {void}
+   */
+  zipConfigFiles: async () => {
+    const fileName = `config-${new Date().toJSON()}.zip`
+    child_process.execSync(`zip -r ${fileName} *`, {
+      cwd: strapi.config.get('plugin.config-sync.syncDir')
+    });
+    const fullFilePath = `${strapi.config.get('plugin.config-sync.syncDir')}${fileName}`
+    const stats = fs.statSync(fullFilePath);
+
+    const result = await strapi.plugins.upload.services.upload.upload({
```

Comment:
Uploading through the upload plugin makes the zip file publically accesable.
That would be a security risk as there may be sensitive data in the `/config/sync` directory.

Is there a way we can download the zip from memory, instead of writing it to the disk?

Reply:
resolved